lib/times.c

```diff
@@ -548,7 +547,9 @@ static int breakdown_time_to_iso8601(const struct timeval *t, struct tm *tm,
     }
     gmtoff /= 60;
 
-    rlen = strftime(buf, len, datefmt, tm);
+    rlen = strftime(buf, len,
+                    withsep ? "%Y-%m-%dT%H:%M:%S" : "%Y%m%dT%H%M%S",
```

Comment:
Heh, yeah or we could use this pattern in a few more places.

Reply:
Yeah this pattern is fine here because it's terse -- there's only two cases, and the format strings are short.  The other spots where I left the variable and shushed the warnings had more cases and/or long format strings, so a ternary inline would've been a mess.